database/rrdcontext.c

```diff
@@ -538,6 +548,36 @@ static void rrdmetrics_destroy(RRDINSTANCE *ri) {
     ri->rrdmetrics = NULL;
 }
 
+static inline bool rrdmetric_should_be_deleted(RRDMETRIC *rm) {
+    if(likely(rrd_flag_is_collected(rm)))
+        return false;
+
+    if(likely(rm->rrddim))
+        return false;
+
+    if(likely(!(rm->flags & RRD_FLAG_DELETED)))
+        return false;
+
+    if(likely(!(rm->flags & RRD_FLAG_LIVE_RETENTION)))
+        return false;
+
+    if(unlikely(rm->flags &
```

Comment:
I will suggest to create a preprocessor like `RRD_FLAGS_PROPAGATED_UPSTREAM` to these 4 flags that we are using here.